applications/asset_tracker/src/main.c

```diff
@@ -988,8 +988,8 @@ static void modem_rsrp_handler(char rsrp_value)
 	 * Checking CONFIG_HOLD_TIME_RSRP gives the compiler a shortcut.
 	 */
 	if (CONFIG_HOLD_TIME_RSRP == 0 ||
-	    k_delayed_work_remaining_get(&rsrp_work) == 0) {
-		k_delayed_work_submit_to_queue(&application_work_q, &rsrp_work,
+	    k_ticks_to_ms_ceil32(k_work_delayable_remaining_get(&rsrp_work)) == 0) {
```

Comment:
What's the value in checking the remaining time?  Just schedule it; if it's already scheduled (remaining is not zero) there will be no change to the existing deadline.

Reply:
done